### PR TITLE
move GCP app/project dns module to here

### PR DESCRIPTION
**Pull request by cmharlow**
Jira: came up in context of https://mozilla-hub.atlassian.net/browse/OPST-333

What this PR does:
* creates a new module for creating GCP Cloud DNS Zone & RecordSet when delegating a subdomain of some parent zone  in mozgcp.net (like testapp.nonprod.dataops.mozgcp.net, a subdomain of dataops.mozgcp.net) to an app-specific GCP Project's own Cloud DNS Zones
* basically replaces: https://github.com/mozilla-services/cloudops-infra-terraform-modules/tree/master/gcp-project-dns

### CODE REVIEW

File: google_project-dns/variables.tf [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    error_message = "Valid values for realm: global, mgmt, nonprod, prod."
+  }
+}
+
+variable "team_name" {
+  description = "Name of SRE team, which should correspond to the top-level folder name"
+  type        = string
+
+  validation {
+    condition     = contains(["cloudops", "dataops", "websre"], var.team_name)
```
Comment (jasonthomas) [inline google_project-dns/variables.tf:37]:
```suggestion
    condition     = contains(["servicessre", "dataops", "websre"], var.team_name)
```

File: google_project-dns/variables.tf [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    error_message = "Valid values for team_name: cloudops, dataops, websre."
```
Comment (jasonthomas) [inline google_project-dns/variables.tf:38]:
```suggestion
    error_message = "Valid values for team_name: servicessre, dataops, websre."
```

It's my understanding that we won't use cloudops, at least for now. We can also just have both cloudops and servicessre here as well.

**Review by jasonthomas — DISMISSED** [review]

File: google_project-dns/variables.tf (hunk 2 above)
Comment (cmharlow) [reply google_project-dns/variables.tf:38]: yeah i'll add both to be safe, thanks!

Comment (cmharlow) [reply google_project-dns/variables.tf]: @jasonthomas give it another review hwen you have a moment, thanks!

**Review by jasonthomas — APPROVED** [review]

**Pull request merged**